text/0000-new-test-waiters.md

```diff
@@ -0,0 +1,248 @@
+import { waitForPromise } from 'ember-test-waiters';
+
+export default class MoreFriendz extends Component {
+  didInsertElement() {
+    waitForPromise(someAsyncWork).then(() => {
+      doOtherThings();
+    });
+  }
+}
+```
+
+This new test waiters system has been through multiple iterations of refinement, and is in use and integrated with the test isolation validation system.
+
+## Rename of `ember-test-waiters` to `@ember/test-waiters`
+
+We should consider renaming the `ember-test-waiters` repository to `@ember/test-waiters`, which would relocate it to the Ember org and scope it more explicitly.
+
+## Compatibility
+
+The `ember-test-waiters` addon is backwards compatible with the old test waiters system, allowing applications and addons to gradually migrate to using the new system.
+
+Specifically:
+
+- `registerWaiter` continues to work, and is not deprecated
+- addons using `ember-test-waiters` work _even if consumed in applications that do not use a new enough `ember-test-helpers` version_ (they won't get additional output via test isolation validation such as test waiter names or stack traces)
+
+The old test waiters system ultimately should be deprecated in its own deprecation RFC.
+
+# How We Teach This
+
+API documentation should be available at `api.emberjs.com`.
+
+## New Testing Section in Ember Guides - "Handling Async"
+
+This new test waiters system should be included in the Ember guide's testing section. Information and examples should be provided to allow users to correctly author asynchronous code that can be correctly managed by the testing system.
+
+Specifically, calling this out in a separate section will allow readers to understand the intent of the test waiters system from a high level, as test waiters apply broadly to all the associated types in the testing section.
+
+There are a few concepts that we should focus on:
+
+1. What patterns of asynchronous code can lead to issues in testing
+1. How to utilize test waiters to address those asynchronous patterns
+1. How to leverage the built-in test waiters utilities
+
+## New Testing Section in Ember CLI Guides - "Handling Async (for Addon Authors)"
+
+The important requirement that addon authors need to know about is that when using `ember-test-waiters`, that package should be added as a `dependency`.
+
+Specifically for addon authors, we want to encourage the use of test waiters to prevent async leaks in tests due to unmanaged async. A special section directed at addon authors should be added to help them understand the value to their consumers by adding test waiters, or providing test helpers via `addon-test-support` integrated with test waiters.
+
+# Drawbacks
+
+- the new test waiters system ships code to production bundles, though this is quite small
+
+# Alternatives
+
+- the existing test waiters system could be left in place
+
+# Unresolved questions
+
+- what is the correct order of operations to roll a change like this out?
```

Comment:
Before merge please remove this line. Under detailed design you can add a section "rollout" which makes clear the proposal is:

* rename the addon
* add it to the default ember-cli blueprint
* there is no schedule for deprecating the old waiters, requires a new RFC

If that is correct this comment is simply an organizational, doesn't require FCP restart.

Reply:
Will do.

Reply:
Done.